plugins/catalog/src/js/repositories/RepositoriesDelete.js

```diff
@@ -0,0 +1,72 @@
+import React from "react";
+
+import { componentFromStream } from "data-service";
+import { Subject } from "rxjs/Subject";
+import { Observable } from "rxjs/Observable";
+import "rxjs/add/operator/combineLatest";
+import "rxjs/add/operator/startWith";
+import "rxjs/add/operator/map";
+import "rxjs/add/operator/catch";
+
+import { deleteRepository } from "./data/repositoriesStream";
+
+import RepositoriesDeleteConfirm from "./components/RepositoriesDeleteConfirm";
+import RepositoriesError from "./components/RepositoriesError";
+
+const getErrorMessage = (response = {}) => {
+  if (typeof response === "string") {
+    return response;
+  }
+
+  return response.description || response.message || "An error has occurred.";
+};
+
+const deleteEvent$ = new Subject();
+const deleteRepository$ = deleteEvent$
+  .switchMap(repository => {
+    return deleteRepository(repository.name, repository.url)
+      .startWith({ pendingRequest: true })
+      .map(result => {
+        return {
+          result,
+          pendingRequest: false
```

Comment:
The semantics of deleteRepository$ are:
- Every time we have an event
   - Create a stream that represents a request to delete that repository
   - starting with pending `{pendingRequest: true}`
   - when that stream finishes, convert (map) the answer to the result plus `{pending: false}`

So if many deletes come, every delete is a full cycle of the above (PENDING, return, NOT_PENDING). That is when `switchMap` comes in because if a new event comes, it drops the previous event. 

One could say that there is only one delete that usually happens, but this is usually an idiom of rxjs to switchMap request streams out of events, so only the latest one actually happens.

Reply:
This would in the future be implemented in graphql, so you would so something like
`deleteEvent$.switchMap(repository => graphqlDeleteMutation(repository))`